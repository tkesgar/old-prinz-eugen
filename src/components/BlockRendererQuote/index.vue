<template>
  <blockquote class="blockquote">
    <p :class="`text-${alignment}`">
      {{ text }}
    </p>
    <footer class="blockquote-footer" v-if="caption">
      {{ caption }}
    </footer>
  </blockquote>
</template>

<script>
export default {
  props: {
    data: {
      type: Object,
      required: true
    }
  },
  computed: {
    text () {
      return this.data.text
    },
    caption () {
      return this.data.caption
    },
    alignment () {
      return this.data.alignment
    }
  }
}
</script>
