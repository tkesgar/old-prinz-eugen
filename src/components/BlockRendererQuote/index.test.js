import { mount } from '@vue/test-utils'
import localVue from '../../utils/test/local-vue'
import BlockRendererQuote from './'

describe('BlockRendererQuote', () => {
  const text = 'Hmm... so you\'re the Commander? How long can you keep me amused? I\'m looking forward to it.'
  const caption = 'Prinz Eugen'
  const alignment = 'left'

  const wrapper = mount(BlockRendererQuote, {
    localVue,
    propsData: {
      data: { text, caption, alignment }
    }
  })

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('should render text', () => {
    expect(wrapper.find('blockquote p').text()).toEqual(text)
  })

  test('should render caption', () => {
    expect(wrapper.find('blockquote footer').text()).toEqual(caption)
  })

  test('should add alignment class', () => {
    expect(wrapper.find(`.text-${alignment}`).exists()).toBe(true)
  })
})

describe('BlockRendererQuote with empty caption', () => {
  const text = 'Hmm... so you\'re the Commander? How long can you keep me amused? I\'m looking forward to it.'
  const caption = ''
  const alignment = 'left'

  const wrapper = mount(BlockRendererQuote, {
    localVue,
    propsData: {
      data: { text, caption, alignment }
    }
  })

  test('should not render caption', () => {
    expect(wrapper.find('blockquote footer').exists()).toBe(false)
  })
})
