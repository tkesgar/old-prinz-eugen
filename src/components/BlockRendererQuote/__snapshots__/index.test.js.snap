// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockRendererQuote should match snapshot 1`] = `
<blockquote
  class="blockquote"
>
  <p
    class="text-left"
  >
    
    Hmm... so you're the Commander? How long can you keep me amused? I'm looking forward to it.
  
  </p>
   
  <footer
    class="blockquote-footer"
  >
    
    Prinz Eugen
  
  </footer>
</blockquote>
`;
